### ISSUE TITLE
The extension no longer works because of an "=" in a return statement

### ISSUE THREAD

**Issue by fjsdalksjfsldkafdskjkdsfjkl** [ZachSaucier/Just-Read#297]
Recently, it appears, the extension has stopped functioning.
Whenever you go to work the extension, it just doesn't work.
Looking in the developer's console, it says the error is line **357**, as you put an "=" after return, putting "return = null" rather than just "return null"
Please fix this soon, I love your extension!

**Comment by nkmathew** [comment]
I'm experiencing the same thing

![](https://i.imgur.com/LIv1EbA.png)

**Comment by Hellothisismespeaking** [comment]
I am also having this issue, please fix it soon! I adore this extention.

**Comment by ZachSaucier** [comment]
I've fixed the issue (I thought I did before pushing to the webstore last night but apparently something didn't work right and it got an old version of the file) but for some reason the web store won't let me update the file. I will continue trying but it may be some time before they allow me to upload the fix for some reason.

So sorry about the error and the delay in getting it fixed but it's outside of my control at this point. I'll keep you posted on the status.

**Comment by ZachSaucier** [comment]
In the mean time, if you want to run the latest, working version of Just Read locally I provide [instructions on how to do so](https://github.com/ZachSaucier/Just-Read/#q-how-can-i-run-just-read-from-a-local-copy-on-my-computer). I recommend switching back to the web store version once this issue has been resolved though.

**Comment by Hellothisismespeaking** [comment]
Thanks so much! It worked, though this did make me realise that I use this extention far too much.

**Comment by ZachSaucier** [comment]
> Thanks so much! It worked, though this did make me realise that I use this extention far too much.

You can never use it too much - that just means web articles suck at being readable 🙂Hence why I had to make this extension in the first place.

**Comment by ZachSaucier** [comment]
No update from Google which is sad but not too surprising their support system is very slow and unreliable. 

I will edit this post with updates going forward to not give subscribers too many emails/notifications.

**Comment by JThees** [comment]
Thanks for this, Zach. Having it not work made me realize how much I've come to depend on it, and need to purchase it, so once Google gets it together, you're about to have another paid customer.

**Comment by ZachSaucier** [comment]
> Having it not work made me realize how much I've come to depend on it, and need to purchase it, so once Google gets it together, you're about to have another paid customer.

Thanks! The [Premium version](https://chrome.google.com/webstore/detail/just-read-premium/dpamdgmnffodphoamchpmbclkpmccjga/reviews) is actually still up and updated.

**Comment by JThees** [comment]
Fantastic. Just acquired. This is such a useful tool, so again, kudos and
thanks!

**Comment by Rosika2** [comment]
Hi Zach,
following your advice to try running Just Read from a local copy on my computer I downloaded the zip-file from https://github.com/ZachSaucier/Just-Read/archive/master.zip , 
unpacked it in a folder named "Just-Read" and enabled developer mode in chrome://extensions/ (chromium-browser).
Then: "Load Unpacked", and selecting the unzipped folder. 

But I get an error-message saying the following:

```
Erweiterungssymbol "icons/icon16.png" kann nicht geladen werden.
Manifest konnte nicht geladen werden.
```
So basically: "icons/icon16.png" cannot be loaded and
maifest could not be loaded.

What can I do now?

Thanks a lot in advance.
Greetings.
Rosika

**Comment by ZachSaucier** [comment]
Hey @Rosika2. What does the file structure look like? Unzipping it looks like this for me:
![image](https://user-images.githubusercontent.com/3966722/75260328-70b89c80-57b7-11ea-8dd3-10d965f77dd9.png)

I didn't have any issue loading it as an unpacked extension.

**Comment by Rosika2** [comment]
Hi and thanks for you very fast response.

The contents of my unzipped folder are thus:

```
rosika@rosika-Lenovo-H520e ~/D/w/D/Just-Read> ll
insgesamt 940K
drwx------ 2 rosika rosika 4,0K Feb 16 15:12 ace/
-rw-rw-r-- 1 rosika rosika 339K Feb 16 15:12 ace.js
-rw-rw-r-- 1 rosika rosika 9,1K Feb 16 15:12 background.js
-rw-rw-r-- 1 rosika rosika 9,1K Feb 16 15:12 banner.png
-rw-rw-r-- 1 rosika rosika 6,7K Feb 16 15:12 banner.svg
-rw-rw-r-- 1 rosika rosika 3,3K Feb 16 15:12 book-icon-outlined.svg
-rw-rw-r-- 1 rosika rosika 2,0K Feb 16 15:12 book-icon-strokes.svg
-rw-rw-r-- 1 rosika rosika 1,1K Feb 16 15:12 bug_report.md
-rw-rw-r-- 1 rosika rosika 107K Feb 16 15:12 content_script.js
-rw-rw-r-- 1 rosika rosika  20K Feb 16 15:12 css.js
-rw-rw-r-- 1 rosika rosika 3,2K Feb 16 15:12 dark-styles.css
-rw-rw-r-- 1 rosika rosika 2,5K Feb 16 15:12 default-styles.css
drwx------ 2 rosika rosika 4,0K Feb 16 15:12 docs/
-rw-rw-r-- 1 rosika rosika 4,7K Feb 16 15:12 EULA.md
drwx------ 2 rosika rosika 4,0K Feb 16 15:12 external-libraries/
-rw-rw-r-- 1 rosika rosika  34K Feb 16 15:12 ext-language_tools.js
-rw-rw-r-- 1 rosika rosika 9,8K Feb 16 15:12 ext-searchbox.js
-rw-rw-r-- 1 rosika rosika  996 Feb 16 15:12 feature_request.md
-rw-rw-r-- 1 rosika rosika 5,7K Feb 16 15:12 hide-segments.css
-rw-rw-r-- 1 rosika rosika  16K Feb 16 15:12 icon128.png
-rw-rw-r-- 1 rosika rosika 3,7K Feb 16 15:12 icon16.png
-rw-rw-r-- 1 rosika rosika 1,5K Feb 16 15:12 icon19.png
-rw-rw-r-- 1 rosika rosika  16K Feb 16 15:12 icon48.png
-rw-rw-r-- 1 rosika rosika  18K Feb 16 15:12 icon64.png
drwx------ 2 rosika rosika 4,0K Feb 16 15:12 icons/
drwx------ 2 rosika rosika 4,0K Feb 16 15:12 ISSUE_TEMPLATE/
-rw-rw-r-- 1 rosika rosika  391 Feb 16 15:12 issue_template.md
drwx------ 2 rosika rosika 4,0K Feb 16 15:12 Just-Read-master/
-rw-rw-r-- 1 rosika rosika 1,4K Feb 16 15:12 manifest.json
-rw-rw-r-- 1 rosika rosika  24K Feb 16 15:12 mode-css.js
-rw-rw-r-- 1 rosika rosika 2,0K Feb 16 15:12 options.css
-rw-rw-r-- 1 rosika rosika 3,2K Feb 16 15:12 options.html
-rw-rw-r-- 1 rosika rosika  19K Feb 16 15:12 options.js
-rw-rw-r-- 1 rosika rosika 1,1K Feb 16 15:12 page.css
-rw-rw-r-- 1 rosika rosika  34K Feb 16 15:12 README.md
-rw-rw-r-- 1 rosika rosika 4,3K Feb 16 15:12 required-styles.css
drwx------ 2 rosika rosika 4,0K Feb 16 15:12 snippets/
-rw-rw-r-- 1 rosika rosika 2,8K Feb 16 15:12 theme-crimson_editor.js
-rw-rw-r-- 1 rosika rosika 138K Feb 16 15:12 worker-css.js

```

**Comment by ZachSaucier** [comment]
@Rosika2 It looks like when you uncompressed the ZIP the directories were flattened? Here's what it should look like (I am using bash so the command is different but functionally equivalent):
```
zs:~/Downloads/Just-Read-master$ ls -alF
total 448
drwxrwxr-x@ 19 zs  staff     608 Feb 25 10:22 ./
drwx------@ 59 zs  staff    1888 Feb 25 10:22 ../
-rw-r--r--@  1 zs  staff    6148 Feb 25 10:22 .DS_Store
drwxrwxr-x@  3 zs  staff      96 Feb 19 14:43 .github/
-rw-rw-r--@  1 zs  staff   34125 Feb 19 14:43 README.md
-rw-rw-r--@  1 zs  staff    9273 Feb 19 14:43 background.js
-rw-rw-r--@  1 zs  staff  108546 Feb 19 14:43 content_script.js
-rw-rw-r--@  1 zs  staff    3260 Feb 19 14:43 dark-styles.css
-rw-rw-r--@  1 zs  staff    2551 Feb 19 14:43 default-styles.css
drwxrwxr-x@  4 zs  staff     128 Feb 19 14:43 docs/
drwxrwxr-x@  3 zs  staff      96 Feb 19 14:43 external-libraries/
-rw-rw-r--@  1 zs  staff    5752 Feb 19 14:43 hide-segments.css
drwxrwxr-x@ 11 zs  staff     352 Feb 19 14:43 icons/
-rw-rw-r--@  1 zs  staff    1382 Feb 19 14:43 manifest.json
-rw-rw-r--@  1 zs  staff    2020 Feb 19 14:43 options.css
-rw-rw-r--@  1 zs  staff    3178 Feb 19 14:43 options.html
-rw-rw-r--@  1 zs  staff   19185 Feb 19 14:43 options.js
-rw-rw-r--@  1 zs  staff    1108 Feb 19 14:43 page.css
-rw-rw-r--@  1 zs  staff    4394 Feb 19 14:43 required-styles.css
```

In other words it should match the structure in this GitHub repo. I'm not sure how you uncompressed it but that seems to be the issue I think.

**Comment by Rosika2** [comment]
Oh, I think I got it. Just a moment please.

**Comment by Rosika2** [comment]
Yes, you´re perfectly right.
When uncompressing the zip I didn´t check "maintain file structure" in file-roller.
Now I did and the structure is maintained, i.e. "icon16.png" is in the icons folder  etc.

Everything works fine now.
It was really my mistake. Sorry for troubling you.
And thanks a lot for your help.

Greetings.
Rosika

**Comment by ZachSaucier** [comment]
I'm glad you got it solved! I'm happy to help.

**Comment by Rosika2** [comment]
@ZachSaucier:
Hello and sorry for troubling you again.
You could help me just fine yesterday and "Just Read" works as it should. So no complaints there.

Yet  there´s just one thing which gives me a bit of a headache.
`chrome://extensions` show the installed add-ons and "Just Read" points out an error:
The following screenshot show the message:




![kgw_Just-Read_error-sprite](https://user-images.githubusercontent.com/37073925/75343011-da0fdc80-5897-11ea-8882-5ede92cc77ce.jpg)

As I said "Just Read" works. So I more or less would like to know if I can ignore this error or if something should/could be done about it.

Thanks again in advance.
Greetings.
Rosika

**Comment by ZachSaucier** [comment]
@Rosika2 You can ignore that warning. It is added in order for the Firefox version of Just Read to work correctly but has no effect in Chrome/Edge 🙂

**Comment by Rosika2** [comment]
@ZachSaucier:
Thank your for your reply.
That´s great. So I can use "Just Read" with reassurance.

Thanks for your great work and support. It´s much appreciated. 

Have a nice day and many greetings.
Rosika

**Comment by hyphenminus** [comment]
Sorry that Google are still delaying the implementation of your fix.

In the meantime, as an easier alternative to manually installing the local version, it might help some people to know that they can instead simply edit the broken file directly in Google's existing installation of the extension.

On Windows, the folder containing the file is typically located somewhere like

- **C:\Users\\*{YOUR_WINDOWS_LOGIN_NAME_HERE}*\AppData\Local\Google\Chrome\User Data\Default\Extensions\dgmanlpmmkibanfdgjocnabmcaclkmod\2.2.3_0\\**

where the file  **content_script.js** can be edited (after saving a backup copy of the uncorrected file, to be cautious!) to replace `return = null;` on line 357 (just after the line about "Invalid JSON schema") with the correct code `return null;`

Just Read then works on next starting Google Chrome.

I assume that editing the file will not affect the next automatic update once Google has eventually approved your revised version for the Chrome Web Store, though it might be safest to make a note of any custom settings before then in case the eventual revised version gets applied afresh.

**Comment by mrtgrizzly** [comment]
Thanks for being on top of this.  I thought my world was coming to an end.  I'm grateful for this extension.  I look forward to the return of its functionality.

**Comment by fadingsignal** [comment]
I went ahead and purchase Premium (which I didn't realize existed.)  Well worth $9/year to actually be able to read websites! :smile: Thanks again @ZachSaucier

**Comment by ZachSaucier** [comment]
Still no update from Google. I wrote a brief post about it that I'm hoping will get some attention. You can upvote it on Hacker News: https://news.ycombinator.com/newest It's titled "Google is blocking me from updating my own Chrome extension".

**Comment by njascgil** [comment]
Version 2.2.4 in online now! Great! Thanks!

**Comment by ZachSaucier** [comment]
Praise God! Thanks for the notification. They don't notify me about it being updated.
